feat(components/modals): improve `SkyModalConfigurationInterface.providers` type

Blackbaud-ErikaMcVey opened the pull request:
BREAKING CHANGE: `SkyModalConfigurationInterface.providers` accepts an array of `StaticProvider`s instead of any value.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/blackbaud/skyux?pullRequest=665) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/blackbaud/skyux?pullRequest=665) before we can accept your contribution.<br/>

blcook223 commented:
Quick note on the change to [`webdrivertest/test/common.js`](https://github.com/blackbaud/skyux/pull/665/files#diff-5):

Without this change, the accessibility checks fail for `tabset`. I felt like the `bypass` rule is not one that an individual component of a page can really be responsible for, but this disables the rule for all accessibility checks. 

I'm not sure if this is what you want, but let me know and I can look at other options if necessary.

blackbaud-sky-savage commented:
**Tests passed.** Automated cross-browser testing via BrowserStack and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 3898e7081b6a979d89e0ea34d6aadc681934a256
Build details: https://travis-ci.org/blackbaud-sky-savage/skyux/builds/141663904

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

Blackbaud-PatrickOFriel commented:
I would be inclined to agree, I think the conditions for bypass generally need to be met by the application, not the individual component

Blackbaud-PatrickOFriel commented:
Approved

Blackbaud-ErikaMcVey committed:
feat(components/modals): improve `SkyModalConfigurationInterface.providers` type

BREAKING CHANGE: `SkyModalConfigurationInterface.providers` accepts an array of `StaticProvider`s
instead of any value.

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/665/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/665/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/665/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/665/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/665/storybooks/pages-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/665/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/665/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/665/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/665/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/665/playground/)

Blackbaud-ErikaMcVey commented:
[AB#2315250](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2315250)
